### PR TITLE
or-tools 8.0 (new formula)

### PR DESCRIPTION
**Pull request by Mizux**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
YES
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
YES
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
YES
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
YES
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
YES
-----
note: Have been tested here:
https://github.com/or-tools/homebrew_or-tools/actions/runs/303951348

note: Abseil commit is part of PR #62803, since we need it I rebased the ortools branch on top of abseil branch thus two commits...
Topic on abseil-cpp c++17
https://discourse.brew.sh/t/abseil-and-c-revision/8641

### CODE REVIEW

File: Formula/abseil.rb [hunk 1]
```diff
@@ -15,10 +15,8 @@ class Abseil < Formula
   depends_on "cmake" => :build
 
   def install
-    ENV.cxx11
```
Comment (SMillerDev) [inline Formula/abseil.rb:18]: Please split this into a separate PR.

File: Formula/or-tools.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  depends_on "osi"
+  depends_on "coinutils"
+
+  def install
+    # CMake based build
+    system "cmake", "-S.", "-Bbuild", *std_cmake_args, "-DUSE_SCIP=OFF", "-DBUILD_SAMPLES=OFF", "-DBUILD_EXAMPLES=OFF"
+    system "cmake", "--build", "build", "-v"
+    system "cmake", "--build", "build", "--target", "install"
+
+    # Produce pkg-config file under cmake/make
```
Comment (SMillerDev) [inline Formula/or-tools.rb:33]: This should be submitted as a patch upstream, we don't make our own pkg-config files.

File: Formula/or-tools.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        const RoutingIndexManager::NodeIndex kDepot(0);
+        RoutingIndexManager manager(42, 8, kDepot);
+        RoutingModel routing(manager);
+        std::cout << "done" << std::endl;
+      }
+    EOS
+    system ENV.cxx, "-std=c++17", "test.cpp",
+                    "-I#{include}", "-L#{lib}", "-lortools",
+                    "-o", "test"
+    system "./test"
```
Comment (SMillerDev) [inline Formula/or-tools.rb:64]: Can we check for output here?

File: Formula/abseil.rb (hunk 1 above)
Comment (Mizux) [reply Formula/abseil.rb:18]: just sent PR #62803

File: Formula/or-tools.rb (hunk 2 above)
Comment (Mizux) [reply Formula/or-tools.rb]: ok removed

File: Formula/or-tools.rb (hunk 3 above)
Comment (Mizux) [reply Formula/or-tools.rb]: replaced with various samples we provide

**Comment by Mizux** [comment]
@SMillerDev so this PR is currently blocked by abseil PR #62803

**Comment by Mizux** [comment]
just rebased on top of master (2ae810e800a2825c849927a695016e83437879b9) -> abseil commit is not needed anymore...

File: Formula/or-tools.rb [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  depends_on "cbc"
+  depends_on "cgl"
+  depends_on "clp"
+  depends_on "coinutils"
+  depends_on "gflags"
+  depends_on "glog"
+  depends_on "osi"
+  depends_on "protobuf"
+
+  # This is a CMake based build
```
Comment (SMillerDev) [inline Formula/or-tools.rb:25]:
```suggestion
```

Comment (Mizux) [reply Formula/or-tools.rb]: Applied your patch
side note: For or-tools we maintain 3 build system (makefile (legacy), CMake (FOSS friendly IMHO), bazel (oss version of the google build system use internally)

**Comment by Mizux** [comment]
@SMillerDev you github patch should have remove the empty line too ^^;, so locally I fixit, rebase -i on master and fixup both patch...

File: Formula/or-tools.rb [hunk 5]
```diff
@@ -0,0 +1,58 @@
+class OrTools < Formula
+  desc "Google's Operations Research tools"
+  homepage "https://developers.google.com/optimization/"
+  url "https://github.com/google/or-tools/archive/v8.0.tar.gz"
+  sha256 "ac01d7ebde157daaeb0e21ce54923a48e4f1d21faebd0b08a54979f150f909ee"
+  license "Apache-2.0"
+  head "https://github.com/google/or-tools/archive/master.zip"
```
Comment (dawidd6) [inline Formula/or-tools.rb:7]:
```suggestion
  head "https://github.com/google/or-tools.git"
```

File: Formula/or-tools.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+  bottle do
```
Comment (dawidd6) [inline Formula/or-tools.rb:9]: Delete this block

File: Formula/or-tools.rb (hunk 5 above)
Comment (Mizux) [reply Formula/or-tools.rb]: done

File: Formula/or-tools.rb (hunk 6 above)
Comment (Mizux) [reply Formula/or-tools.rb]: done

**Review by dawidd6 — APPROVED** [review]
